hwy/ops/x86_128-inl.h

```diff
@@ -4254,6 +4264,83 @@ HWY_API Vec128<uint64_t, N> RotateRight(const Vec128<uint64_t, N> v) {
 #endif
 }
 
+// I8/I16/I32/I64 RotateRight is generic for all vector lengths
+template <int kBits, class V, HWY_IF_SIGNED_V(V)>
+HWY_API V RotateRight(V v) {
+  const DFromV<decltype(v)> d;
+  const RebindToUnsigned<decltype(d)> du;
+  return BitCast(d, RotateRight<kBits>(BitCast(du, v)));
+}
+
+// ------------------------------ Rol/Ror
+#if HWY_TARGET <= HWY_AVX3
+
+#ifdef HWY_NATIVE_ROL_ROR_32
+#undef HWY_NATIVE_ROL_ROR_32
+#else
+#define HWY_NATIVE_ROL_ROR_32
```

Comment:
Would it make sense to combine the 'have native' for 8/16, and 32/64?

Reply:
It makes sense to keep HWY_NATIVE_ROL_ROR_8 and HWY_NATIVE_ROL_ROR_16 separated as the U8 Rol/Ror operations but not the U16 Rol/Ror targets might possibly get re-implemented on AVX3_DL targets.

Similarly, it makes sense to keep HWY_NATIVE_ROL_ROR_SAME_8 and HWY_NATIVE_ROL_ROR_SAME_16 separated as the U8 RotateLeftSame/RotateRightSame but not the U16 RotateLeftSame/RotateRightSame might possibly be re-implemented on AVX3 targets to take advantage of ternary logic or AVX3_DL GaloisAffine operations.

Reply:
Thanks for combining 32/64! Right, the GFNI is separate for 8-bit, makes sense.